listenbrainz/webserver/static/css/recommendation-page.less

```diff
@@ -45,16 +45,109 @@
 
       .recommendation-controls {
 
+        > svg:hover {
+          cursor: pointer;
+        }
+
         &.open .dropdown-menu {
           display: flex;
+
+          > svg:hover {
+            cursor: pointer;
+          }
         }
 
-        .smile, .laugh, .angry, .disappoint, .meh {
-          font-size: 18px;
+        .fa-angry, .fa-frown, .fa-meh, .fa-smile-beam, .fa-grin-stars, .fa-ellipsis-v {
```

Comment:
We shouldn't be using the fontawesome classnames here, because it makes it hard to replace them with something else in the future (or with userscripts I guess).

Instead, each feedback item should have its own classname that you can target, the way you have it below like `&.angry{…`.
So instead of `.fa-angry { &.angry …` I think it would be better to have something like `.angry { &.active …` or `.angry { &.selected …` or some other word to denote that that particular feedback has been selected.

Reply:
right, I agree :)

Reply:
So a couple of things here, I guess I am a bit confused:
 ```{`btn ${currentFeedback}`}``` is the button class which changes with feedback
The CSS linked:
```
.btn {
          font-size: 18px;
          color: @blue;
          background-color: white;

          &.hate {
            color: #C62828;
          }

          &.dislike {
            color: #f5873d;
          }

          &.bad_recommendation {
            color: #ebbf0b;
          }

          &.like {
            color: #b0e111;
          }

          &.love {
            color: #32cd32;
          }

          &:hover {
            background-color: @blue;
            color: white;
          }
        }
```
so if feedback == hate, the class will be `className='btn hate`, the colour of the icon will change to red and so on

The CSS for the 5 emoticons. This is not linked with any class as such. It kinds defines the base property of the emotions. 

```
.fa-angry, .fa-frown, .fa-meh, .fa-smile-beam, .fa-grin-stars, .fa-thumbs-up {
          stroke-width: 40px;
          font-size: 18px;
          flex: 1;
          text-align: center;
          margin: 0 0.4em;

        }
        .fa-angry {
          color: #C62828;

        }

        .fa-frown {
          color: #f5873d;

        }

        .fa-meh {
          color: #ebbf0b;

        }

        .fa-smile-beam {
          color: #b0e111;

        }

        .fa-grin-stars {
          color: #32cd32;

        }
```
Our goal is to not use the icon names as it is or `.fa-angry, .fa-frown, .fa-meh, .fa-smile-beam, .fa-grin-stars, .fa-thumbs-up`
if we have to use something else other than the names of the icons, we will need to link that to some class since we aren't using the default class, I guess. I just understand the problem statement here, I am not really clear about the solution though.

Reply:
Backtracking on what I said above (again!), sorry about the flip-flopping. As the components change so do the solutions :)

I suggested using the `currentFeedback` as a css class for styling.
That makes styling the feedback button (not the dropdown) pretty easy, but we need to extend it to styling the emoticons in the dropdown.

For the common styling (what affects all emoticons like size, margins, etc.) we can use the recommendation-icon class we hardcoded in `RecommendationControl`.
For styling each emoticon separately (for colours), I propose replacing the `selected` property with a `cssClasses` property and setting it like this in `RecommendationCard`:

```
<RecommendationControl
	iconHover={faGrinStars}
	icon={faGrinStarsRegular}
	title="I really love this!"
	action={() =>
	currentFeedback === "love"
		? this.deleteFeedback()
		: this.submitFeedback("love", faGrinStars, "love")
	}
	cssClass={`love ${currentFeedback === "love" ? "selected" : ""}`}
/>
```

You'll notice we are moving the logic of deciding whether to have the 'selected' class applied from the RecControl to the RecCard component.
In RecControl, we'll simply have this:

```
const { cssClass, iconHover, icon, action, title } = props;
  return (
    <div
      className={`recommendation-icon ${cssClass}`}
```

We'll end up with something like this in the page:
<img width="300" alt="Capture d’écran 2020-11-02 à 12 19 23" src="https://user-images.githubusercontent.com/6179856/97862476-b6921100-1d05-11eb-96ed-4b5e64b9c16a.png">

From here it should be pretty easy to style the buttons accordingly:
Here's the entirety of my `.recommendation-control` in the LESS file:

```
.recommendation-controls {

        &.open .dropdown-menu {
          display: flex;
        }

        .btn {
          font-size: 18px;
          color: @blue;
          background-color: white;

          &:hover {
            background-color: @blue;
            color: white;
          }
        }
        
        .hate {
          color: #C62828;
        }

        .dislike {
          color: #f5873d;
        }

        .bad_recommendation {
          color: #ebbf0b;
        }

        .like {
          color: #b0e111;
        }

        .love {
          color: #32cd32;
        }
        
        .recommendation-icon > span {
          stroke-width: 40px;
          font-size: 18px;
          flex: 1;
          text-align: center;
          margin: 0 0.4em;
        }

        .recommendation-icon:hover .off,
        .recommendation-icon.selected .off,
        .recommendation-icon .on {
          display: none;
        }
        .recommendation-icon:hover .on,
        .recommendation-icon.selected .on {
          display: inline;
        }

      }
```

Note that we're only defining the combination [feedback string class -> color property] once, and they apply to both the 'like' button states and the colors of the emoticons in the dropdown.

Reply:
This takes away the drop-down emoticon colours. looking into it.